src/evaluatorHelpers.ts

```diff
@@ -324,29 +336,164 @@ export async function renderPrompt(
   }
 }
 
+// ================================
+// Extension Hooks
+// ================================
+
+// TODO(chore): Move the extension hooks logic into a separate file.
+
+const BeforeAllExtensionHookContextSchema = z.object({
+  suite: TestSuiteSchema,
+});
+
+const BeforeEachExtensionHookContextSchema = z.object({
+  test: TestCaseSchema,
+});
+
+type BeforeAllExtensionHookContext = z.infer<typeof BeforeAllExtensionHookContextSchema>;
+type BeforeEachExtensionHookContext = z.infer<typeof BeforeEachExtensionHookContextSchema>;
```

Comment:
I think you should try to cut as much of this out as possible. This can just use the TestCase type which is already defined in src/types/index.ts. It is harder for me to grok this way